src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private T[] array;
+    private int size = 0;
+
+    public ArrayList() {
+        array = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            T[] newArray = (T[]) new Object[array.length + (array.length / 2)];
+            for (int i = 0; i < size; i++) {
+                newArray[i] = array[i];
```

Comment:
Consider using `System.arraycopy()` instead of manually copying elements. This will improve performance and readability .